modules/frontend/querysharding.go

```diff
@@ -214,9 +214,10 @@ func mergeResponses(ctx context.Context, marshallingFormat string, rrs []Request
 
 		span.SetTag("response marshalling format", marshallingFormat)
 		return &http.Response{
-			StatusCode: http.StatusOK,
-			Body:       ioutil.NopCloser(bytes.NewReader(combinedTrace)),
-			Header:     http.Header{},
+			StatusCode:    http.StatusOK,
+			Body:          ioutil.NopCloser(bytes.NewReader(combinedTrace)),
+			ContentLength: int64(len(combinedTrace)),
```

Comment:
it feels a bit weird to set this. i think that when this http response is returned from the main handler it will be overwritten.

let's put a comment explaining that we're using this to just pass body size up to the method above. not ideal, but better than other options i think.